### PR TITLE
Put importance sampling into no grad

### PR DESCRIPTION
**Pull request by pluesclues**
https://github.com/huggingface/trl/blob/41c8ca1ad3e03ecdd2833874a7eb36e85f31a089/trl/trainer/grpo_trainer.py#L1525-L1529

variable shouldn't have gradients attatched

### CODE REVIEW

File: unsloth_zoo/rl_replacements.py [hunk 1]
```diff
@@ -626,7 +626,14 @@ def grpo_accumulated_loss(
...
             ).logits
-
+            
+    # Ensure old/ref states never carry grads across micro-batches
+    if old_hidden_states is not None and torch.is_tensor(old_hidden_states):
+        old_hidden_states = old_hidden_states.detach()
+    
+    if ref_hidden_states is not None and torch.is_tensor(ref_hidden_states):
+        ref_hidden_states = ref_hidden_states.detach()
+
```
Comment (danielhanchen) [inline unsloth_zoo/rl_replacements.py]:
```suggestion
```

File: unsloth_zoo/rl_replacements.py [hunk 2]
```diff
@@ -626,7 +626,7 @@ def grpo_accumulated_loss(
                 image_sizes = image_sizes,
                 logits_to_keep = logits_to_keep + 1,
             ).logits
-
+
```
Comment (danielhanchen) [inline unsloth_zoo/rl_replacements.py]:
```suggestion

```
